### DISCUSSION TITLE
Why useSWR does not initially fetch if `revalidateOnMount` is set to false?

### DISCUSSION THREAD

**Discussion by jeromevvb** [vercel/swr#2172] [Q&A]
Hey guys,

First of all, thank you for this amazing library!
I have spent one hour trying to understand what i feel its bug but its actual an unexpected behavior in my opinion.

I have `SWRConfig` into my `_app` file (i'm using NextJS) with `revalidateOnMount:false` because i do not want my components to revalidate on mount. I just want them to validate once, right?
So i naturally set `revalidateOnMount` to set to false.

```javascript
const SWRConfigOptions = {
  fetcher: (url: string) => axios.get(url).then((res) => res.data),
  revalidateOnMount: false,
  revalidateOnFocus: false,
  revalidateOnReconnect: true,
  shouldRetryOnError: true,
  errorRetryInterval: 2000,
  errorRetryCount: 2,
};

<SWRConfig value={SWRConfigOptions}>
    <Component {...pageProps} />
 </SWRConfig>
```

Now into my component, i will want to request my api.

```javascript
const Users = () => {
  // fetch users
  const { data,  isValidating } = useSWR(`/api/users`, {
 // i have to manually set to true to make it works, why?
  // revalidateOnMount: true,
  });
})
```

This won't work if don't set manually `revalidateOnMount` to `true`. 
This is extremely confusing for me, i do not want to constantly request my API.
I was expected to fetch at least the first time (to fill up the cache) and then it wont fetch again because i have configured it like it.

Can you guys tell me why ? Am i doing something wrong?
PS: i only want to use client data fetching.

Thank you so much for your help!

**Comment by koba04** [comment]
In your case, you can use the `useSWRImmutable` hook, which doesn't revalidate the data if it has been already cached.
https://swr.vercel.app/docs/revalidation#disable-automatic-revalidations

**Comment by jeromevvb** [comment]
Thank you for your answer @koba04 
Although, `useSWRImmutable` does not solve my confusion.

Even with useSWRImmutable, it wont fetch the first time if `revalidateOnMount` is not set to `true`
I just think this name lead to confusion, since it acts as if it was `validateOnMount`

**Reply by koba04** [reply]
What about using useSWRImmutable with revalidateOnMount: undefined?
